[docs] Add BugSnag guide (#26987)
expo/expo: docs/pages/guides/using-bugsnag.mdx

---
title: Use BugSnag
description: A guide on installing and configuring BugSnag for end-to-end error reporting and analytics.
hideTOC: true
---

import { BoxLink } from '~/ui/components/BoxLink';
import { BookOpen02Icon } from '@expo/styleguide-icons';

> [BugSnag](https://www.bugsnag.com) is a stability monitoring solution that provides rich, end-to-end error reporting and analytics to reproduce and fix errors with speed and precision. BugSnag supports the full stack with open-source libraries for more than[50 platforms](https://www.bugsnag.com/platforms), including [React Native](https://docs.bugsnag.com/platforms/react-native/react-native/).

Reviewer: ```suggestion
[BugSnag](https://www.bugsnag.com) is a stability monitoring solution that provides rich, end-to-end error reporting and analytics to reproduce and fix errors with speed and precision. BugSnag supports the full stack with open-source libraries for more than [50 platforms](https://www.bugsnag.com/platforms), including [React Native](https://docs.bugsnag.com/platforms/react-native/react-native/).
```
missing space

With BugSnag, developers and engineering organizations can:

- **Stabilize:** Innovate faster by knowing when to build new features versus fix bugs. Use the release health dashboard, stability scores and targets, and built-in alerts via email, Slack, PagerDuty, and more.
- **Prioritize:** Improve customer experience by identifying and prioritizing bugs that have the greatest impact on app stability. Analyze issues grouped by root cause and sorted by business impact, customer segmentation, A/B testing and experiment results.
- **Fix:** Increase productivity by spending less time on reproducing and fixing bugs. Utilize powerful diagnostic data, full stacktraces and automatic breadcrumbs.

## Integration

> See the integration guide below for instructions on how to add BugSnag to your Expo apps to report JavaScript errors. This guide includes instructions for uploading source maps for updates published with [EAS Update](/eas-update/introduction/).

Author: ```suggestion
See the integration guide below for instructions on adding BugSnag to your Expo apps to report JavaScript errors. It also includes instructions for uploading source maps for updates published with [EAS Update](/eas-update/introduction/).
```

> See the integration guide below for instructions on how to add BugSnag to your Expo apps to report JavaScript errors. You can also read the [blog post](https://www.bugsnag.com/blog/build-apps-in-expo-with-bugsnag) announcing BugSnag's Expo integration. 

Reviewer: ```suggestion
See the integration guide below for instructions on how to add BugSnag to your Expo apps to report JavaScript errors. This guide includes instructions for uploading source maps for updates published with [EAS Update](/eas-update/introduction/).
```
The blog link is dead, so suggested removing that.

That they do include info about EAS Update support seems relevant to users arriving here to see what bug reporting options they have. It's included in the below link, but if we wanted to link to that specifically, it's here: https://docs.bugsnag.com/platforms/react-native/expo/#eas-update
